From 3124282ec4dd89decd2700802da37c787ee610a5 Mon Sep 17 00:00:00 2001
From: Demi Marie Obenour <demi@invisiblethingslab.com>
Date: Tue, 23 Mar 2021 11:48:02 -0400
Subject: [PATCH 06/22] Require signature header to be a contiguous region

This reduces attack surface and is compatible with all packages
generated by upstream RPM.  Upstream changed this for compatbility
reasons that do not affect QubesOS.
---
 lib/rpmchecksig.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/rpmchecksig.c b/lib/rpmchecksig.c
index 6bc6a61..f57dcfa 100644
--- a/lib/rpmchecksig.c
+++ b/lib/rpmchecksig.c
@@ -171,8 +171,8 @@ rpmRC rpmpkgRead(struct rpmvs_s *vs, FD_t fd,
 	goto exit;
     }
 
-    /* Read the signature header. Might not be in a contiguous region. */
-    if (hdrblobRead(fd, 1, 0, RPMTAG_HEADERSIGNATURES, sigblob, &msg))
+    /* Read the signature header. Must be in a contiguous region. */
+    if (hdrblobRead(fd, 1, 1, RPMTAG_HEADERSIGNATURES, sigblob, &msg))
 	goto exit;
 
     rpmvsInit(vs, sigblob, bundle);
-- 
2.30.2

